runtime/common/src/zenlink.rs

```diff
@@ -0,0 +1,272 @@
+use spacewalk_primitives::{Asset, CurrencyId};
+
+use zenlink_protocol::{LOCAL, NATIVE};
+pub type ZenlinkAssetId = zenlink_protocol::AssetId;
+
+pub const USDC_ISSUER: [u8; 32] = [
+	59, 153, 17, 56, 14, 254, 152, 139, 160, 168, 144, 14, 177, 207, 228, 79, 54, 111, 125, 190,
+	148, 107, 237, 7, 114, 64, 247, 246, 36, 223, 21, 197,
+];
+
+pub const BRL_ISSUER: [u8; 32] = [
+	234, 172, 104, 212, 208, 227, 123, 76, 36, 194, 83, 105, 22, 232, 48, 115, 95, 3, 45, 13, 107,
+	42, 28, 143, 202, 59, 197, 162, 94, 8, 62, 58,
+];
+
+pub const TZS_ISSUER: [u8; 32] = [
+	52, 201, 75, 42, 75, 169, 232, 181, 123, 34, 84, 125, 203, 179, 15, 68, 60, 76, 176, 45, 163,
+	130, 154, 137, 170, 27, 212, 120, 14, 68, 102, 186,
+];
+
+fn discriminant(currency: &CurrencyId) -> u8 {
+	match currency {
+		CurrencyId::Native => 0,
+		CurrencyId::XCM(_) => 1,
+		CurrencyId::Stellar(_) => 2,
+		CurrencyId::ZenlinkLPToken(_, _, _, _) => 6,
+	}
+}
+
+pub fn zenlink_id_to_currency_id(asset_id: ZenlinkAssetId, parachain_id: u32) -> Result<CurrencyId, ()> {
```

Comment:
Yes, that's true. I think it would be a good idea to verify that the asset type is correct. After completing the entire bootstrap process, I realized that we manually set the asset indexes, asset types and chain id. To avoid errors, it would be a good idea to check that we have set the asset type correctly.